Refactor plugin details modal using Vue

michalkleiner opened the pull request:
### Description:

Ref. DEV-17864 and DEV-17371

Refactor plugin details view from back-end twig template to Vue template so that Start Free Trial action can be integrated into the modal.

Implement new modal design with sticky header and footer. CTA action in the modal is the same as on the card (just without the More details that would open the modal itself).

The legacy modal and its CSS was kept unchanged.

Once initial review is done we can think of adding UI tests for the new modal.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

michalkleiner committed:
Add translations

michalkleiner committed:
Pass 'has some admin access' down to components

michalkleiner committed:
Allow CTA container to optionally only display relevant actions (and not notifications or links)

michalkleiner committed:
Pass number of users down to view components

michalkleiner committed:
Reimplement plugin details modal in Vue

mneudert commented on plugins/Marketplace/vue/src/PluginDetails/PluginDetails.vue:
```diff
@@ -0,0 +1,448 @@
+      return filename.substring(0, filename.lastIndexOf('.')).split('_').join(' ');
+    },
+    showPluginDetailsDialog() {
+      $('#pluginDetails').modal({
```
The new materialize modal does not seem to stretch/overflow nicely:

![](https://github.com/matomo-org/matomo/assets/635801/e9775b85-f10f-4557-b6ab-2cd9f9a2d39b)
![](https://github.com/matomo-org/matomo/assets/635801/9aadd5f1-9437-4fd7-88cd-44b9350223b1)
![](https://github.com/matomo-org/matomo/assets/635801/9f431c86-3226-4670-ac74-bac0d329b11d)

Depending on the viewport and modal content the scrollbars are all over the place.

michalkleiner replied:
What browser was that in? It was working ok for me in Brave browser.

mneudert replied:
The screenshots were from a current desktop Chrome. My current Firefox has less issues with the scrollbars, but the vertical overflow on the right column is also visible there at like 800px viewport height.

michalkleiner replied:
I'm on a Mac so can't test other operating systems (Linux, Windows) so would appreciate any cooperation on this :)

mneudert replied:
It looks like `#pluginDetails { overflow-y: initial }` is part of the problem. Removing it will make the complete modal scrollable, probably an undesired state.

But with `overflow: hidden` we face the issue of the parts itself not being scrollable. Throwing around `height: 100%` on `.modal_content, .pluginDetails, .row, .col` (with an additional `overflow-y: scroll` on `.col`) made it partially scrollable, yet still horrible, because the `<h2>` outside of `.pluginDetails` is messing with the total space available and pushes things down.

In my local browsers it looks like we have to override everything to flexbox so much that it may be easier to just remove `.row / .col` and build it fresh.

michalkleiner replied:
In that case it might be best to spend the time and energy and refactor the markup into the final state as per the phase 2 designs, I don't think refactoring the markup and styles to achieve the same is the best use of the time.

mneudert commented on plugins/Marketplace/stylesheets/plugin-details.less:
```diff
@@ -102,5 +97,39 @@
+#pluginDetails {
+    overflow-y: initial;
+    height: 70%;
+    width: 55%;
```
The 55% width get really small really quick below 1600px viewport size, some responsive height/width adjustment could be a good thing until the complete modal rework is done.

michalkleiner replied:
Yeah, I kept it at 55 to keep it as it is for the current modal until the rework. I was playing with 70 but then the right column gets too wide with the buttons.

michalkleiner replied:
I'll mark this as resolved for now, we can always iterate on this if product deems so.

mneudert commented on plugins/Marketplace/vue/src/PluginDetails/PluginDetails.vue:
```diff
@@ -0,0 +1,448 @@
+                <template v-if="plugin.numDownloads">
+                <dt>{{ translate('General_Downloads') }}</dt>
+                <!--                number format below-->
+                <dd title="{{ translate('Marketplace_NumDownloadsLatestVersion', pluginLatestVersion.numDownloads) }}">{{ plugin.numDownloads }}</dd>
```
The translation for the tooltip is currently missing on the client side.

mneudert commented on plugins/Marketplace/vue/src/PluginDetails/PluginDetails.vue:
```diff
@@ -0,0 +1,448 @@
+                </div>
+
+                <div id="tabs-description" class="tab-content col s12">
+                  <div v-if="showMissingRequirementsNoticeIfApplicable" vue-entry="Marketplace.MissingReqsNotice" plugin="{{ plugin|json_encode }}"></div>
```
```suggestion
                  <MissingReqsNotice v-if="showMissingRequirementsNoticeIfApplicable" :plugin="plugin" />
```

_Also requires the matching import for it._

mneudert commented on plugins/Marketplace/vue/src/PluginDetails/PluginDetails.vue:
```diff
@@ -0,0 +1,448 @@
+                  <div v-if="showMissingLicenseDescription" class="alert alert-danger">{{ translate('Marketplace_PluginLicenseMissingDescription') }}</div>
+                  <div v-else-if="showExceededLicenceDescription" class="alert alert-warning">{{ translate('Marketplace_PluginLicenseExceededDescription') }}</div>
+
+                  <div v-html="$sanitize(pluginDescription)"></div>
```
Something odd is happening:

- open modal for plugin
- close modal
- reopen modal for same plugin
- several tabs are gone/empty

:monocle_face:

michalkleiner replied:
ah yep, could replicate this.. weird, will investigate

michalkleiner replied:
Resolved, I was incorrectly removing the latest version and then there were none and no info to display.

mneudert commented on plugins/Marketplace/vue/src/PluginList/CTAContainer.vue:
```diff
@@ -6,23 +6,25 @@
     </div>
 
-    <div v-else-if="plugin.hasExceededLicense"
+    <div v-else-if="!showActionsOnly && plugin.hasExceededLicense"
```
We may need to adjust the conditions somewhat. Opening the details modal for `Sentry` plugin with a PHP 7.2 install shows the `Install` button, but with an unmatched requirement of `PHP >= 7.4` it should probably be hidden (have not really checked, but I think this was the case before).

michalkleiner replied:
Will check this.

michalkleiner replied:
This has been removed in favour of the CTA behaving the same on the card and in the modal.

mneudert commented on plugins/Marketplace/vue/src/PluginList/CTAContainer.vue:
```diff
@@ -77,16 +79,18 @@
     >{{ translate('Marketplace_StartFreeTrial') }}</a>
```
We should reuse this event to close the plugin details modal, otherwise the various modals are displayed on top of each other:

![](https://github.com/matomo-org/matomo/assets/635801/faea0975-2c3a-4e1e-b154-d86e1358deca)

michalkleiner replied:
I was looking into that but my preference was to keep both open as it's then hard to go back to the detail modal if the free trial modal was opened from details modal.

That's probably a question for product/design, but for now I'd keep it like it is.

mneudert replied:
Let's have product/design decide if we need to reiterate on a later date :+1: 

We only need to make sure the modal is getting closed with the Marketplace reload when the trial is actually started. But that should not be much of a problem as the whole component should be destroyed and recreated I would assume :thinking:

michalkleiner replied:
@ronak-innocraft @Javi-Ormaechea what would you expect to happen here when opening the "Start free trial" modal from the details modal? Should they overlap or should the details close first? If we close the details first, we currently don't have a way to open it back when the "Start free trial" is closed.

Javi-Ormaechea replied:
@michalkleiner what do you mean by 'we currently don't have a way to open it back when the start free trial is closed?' can users not click on it again if they wanted to view the plugin details? 

Best practice suggests that having a modal on a modal it's not the best experience.

Keeping both as @michalkleiner said allows users to close it and stay on the plugin details if they don't want to proceed. On the other hand once a user actively clicks on start a free trial they are making the decision to try the plugin so we can assume that the details are no longer needed as they have made up their mind. Valid questions here are; if they didn't want to proceed do they actually need to go back to the details? Equally, maybe they want to check something else before proceeding?

So based on assumptions (and the above question being yes), my suggestion is to stick to best practices and don't show a modal on a modal.

michalkleiner replied:
@Javi-Ormaechea I meant go back to the details automatically if they close the start free trial modal.

So currently it is like this (without a licence):
1. details modal opens
2. user clicks start trial
3. missing licence modal opens
4. user closes missing licence modal
5. user still sees details modal, they are where they were

If we close the details modal before showing the missing licence modal:
1. details modal opens
2. user clicks start trial
3. details modal closes
4. missing licence modal opens
5. user closes missing licence modal
6. user lost which plugin details they were looking at

I can see if there's a way how to reopen the details modal if it was closed due to the start trial action and possibility to go back to the plugin details automatically.

michalkleiner replied:
As discussed internally, it's desired not to have a modal on top of a modal, so we need to close the details modal before opening the start trial modal.

Extra point is possibly open back the details if it was the initiator for the start trial modal, but it's not critical.

michalkleiner replied:
The modal now closes before the Start trial modal is shown.

mneudert commented on plugins/Marketplace/Controller.php:
```diff
@@ -156,41 +156,6 @@ private function getPrettyLongDate($date)
         return Date::factory($date)->getLocalized(Date::DATE_FORMAT_LONG);
     }
 
-    public function pluginDetails()
```
It seems it is not as easy as hoped. The `CoreAdminHome/PluginName.ts` directive is opening the old modal using this method. And it is used in various places like "General Settings" on the "Premium Features" card, outside of Marketplace :(

In the Marketplace at least the "Help" link is also using this directive, but that should be an easy fix.

michalkleiner replied:
Yeah 🤯

I guess we can keep the method for now for places outside of Marketplace.

mneudert replied:
But that also means we need to keep the Twig template and update it to make use of the `CTAContainer` vue component (and hopefully only that one) :(

michalkleiner replied:
That feels like a massive PITA, but I can try as the last thing. I would rather revisit and remove links to the plugin detail modal from other places.

mneudert replied:
It will definitely be painful... as in: if a trial is started, we display a notification, but we cannot reload marketplace because marketplace is not available? :exploding_head:

Maybe we can rework it to have a link to marketplace (and perhaps re-open the modal there directly) instead of the CTA? But that sounds like a weird workflow...

Or maybe link to the open modal in Marketplace. That could work for the "Premium Features" card. But then the titles on the "Manage Plugins" page also open those modals :thinking:

michalkleiner replied:
How about the modal not offering to start a trial for places outside of Marketplace? Basically keep the current behaviour for outside of marketplace, until we figure out how to do that, maybe during the redesign phase?

michalkleiner replied:
It will mean for a while we would have two slightly differently looking/behaving modals.

mneudert replied:
That might be the best way to get around this issue temporarily :thinking:

michalkleiner replied:
As discussed internally new issue will be lodged for this. Until resolved then we're keeping the method and the template.

michalkleiner committed:
Use MissingReqsNotice component directly

michalkleiner committed:
Remove crypto dependency as simpler mechanism can be used

michalkleiner committed:
Add target blank to content links

michalkleiner committed:
Fix getting latest version mechanism to prevent tabs breaking

Incorrect use of pop removed the last version from the list and it was not available for subsequent use.

michalkleiner committed:
Add types

michalkleiner committed:
Use root reference when filtering for matching elements inside setTimeout callback

michalkleiner commented:
Moving to draft as parked internally for now.

michalkleiner committed:
Hide More details button from plugin CTA when displayed in modal

Remove showActionsOnly flag and replace it with inModal flag as PO agreed to display the same actions
in the modal as on the plugin card, except the More details button which would open the modal so it
wouldn't make sense to display that.

michalkleiner committed:
Implement plugin details modal redesign

michalkleiner committed:
Make links from URLs and emails in the extra support information

michalkleiner commented:
@mneudert I'd appreciate another review — the markup for the modal was refactored to use flexbox with a fixed header and footer and only the content should be scrolling now. For other decisions please refer to JIRA DEV-17371, e.g. keeping the CTA behaviour the same as on the plugin card, the discussion around the featured plugin (for now next to the title) or the removal of the promo content section (not used).

mneudert commented on plugins/Marketplace/vue/src/PluginDetailsModal/PluginDetailsModal.vue:
```diff
@@ -0,0 +1,508 @@
+      const licence: TObject = this.pluginLatestVersion?.license as TObject || {};
+      return !!licence.name;
+    },
+    showPluginVariations(): boolean {
```
Looks unused, matching the `numUsers` removable? Or are we actually missing the variant selection and a CTA?

michalkleiner replied:
Correct, we don't need this, it was a leftover from the previous iteration before we decided to drop the variant selection (as per recorded decision in JIRA).

Removed.

mneudert commented on plugins/Marketplace/vue/src/PluginDetailsModal/PluginDetailsModal.vue:
```diff
@@ -0,0 +1,508 @@
+      return this.isSuperUser && (this.plugin.isDownloadable || this.plugin.isInstalled);
+    },
+    showLicenceName(): boolean {
+      const licence: TObject = this.pluginLatestVersion?.license as TObject || {};
```
We should probably change all variables/functions to use `license`, to be consistent with the one naming that may be harder to change :sweat_smile:

michalkleiner replied:
Good catch, that was an oversight. Changed to `license`.

mneudert commented on plugins/Marketplace/vue/src/PluginDetailsModal/PluginDetailsModal.vue:
```diff
@@ -0,0 +1,508 @@
+      type: Boolean,
+      required: true,
+    },
+    numUsers: {
```
Variable is not used anywhere, probably from pre auto-user-count-trial design days?

michalkleiner replied:
Good catch, will remove the number of users from the Marketplace and nested components as the product variant selection for free trials is automatic in the backend and we dropped the dropdowns where this was used to select the current applicable tier.

mneudert replied:
One usage left in the Controller and the main `overview.twig`, looks like we can remove those as well.

michalkleiner replied:
Must've mistaken it for the one for the plugin details that I wanted to keep in this PR. 
Removed.

mneudert commented on plugins/Marketplace/vue/src/PluginDetailsModal/PluginDetailsModal.vue:
```diff
@@ -0,0 +1,508 @@
+                      :href="author.homepage"
+                    >{{ author.name }}</a>
+                    <a
+                      v-else-if="author.email"
```
Should this also be wrapped in an `isValidEmail` check?

michalkleiner replied:
Yep, have added the new email validity check here as well. Originally it didn't have it in the legacy modal but it makes sense to have it here as well.

mneudert commented on plugins/Marketplace/vue/src/PluginDetailsModal/PluginDetailsModal.vue:
```diff
@@ -0,0 +1,508 @@
+                    </dd>
+                    <dd v-else-if="this.isValidEmail(support.value)">
+                      <a
+                        :href="`mailto:${$sanitize(support.value)}`"
```
Is there some wild encoding going on that requires `$sanitize()` here while for the authors it is `encodeURIComponent()`? (or the other way round of course)

michalkleiner replied:
I went with what was originally in the template and these data points could technically be any text so I guess that's why it used `$sanitize` compared to just `encodeURIComponent` for the email where it was probably more likely only an email than anything else. I guess we can unify it around the email since the check for valid email should already be enough, and keep `$sanitize` for other supporting values.

mneudert commented on plugins/Marketplace/vue/src/types.ts:
```diff
@@ -0,0 +1,99 @@
+  numDownloadsPretty: TNumberOrString;
+}
+
+export interface PluginDetailsState {
```
I have usually seen the state being defined alongside the component itself. I don't think it will be reused anywhere so we don't need it importable , or was there an intent in moving this here?

michalkleiner replied:
No, it was just that I put all types into the types file for the plugin, no explicit/specific intentions other than to keep the length of the files smaller.

Will move the state back to the component as it's quite small in this case.

mneudert commented on plugins/Marketplace/stylesheets/marketplace.less:
```diff
@@ -345,6 +345,10 @@
     }
+
+    .modal {
+        background-color: #fff;
```
We probably should double check with design how global that background color should be. 

Currently it would only affect marketplace, other places (e.g. user manager password confirmation) would still be displayed with the default `background-color: #fafafa;`.

Or was the intention to only have it for the details and we just move it there?

michalkleiner replied:
Yes, it is intentionally here as the design has white background. I've seen some white artefacts around elements on the grey background, though that was with before the layout redesign. I can see if we can go to `#fafafa` background and we can also ask design team what should the modal background be globally.

mneudert replied:
Will create an internal task to verify the background color. Will update the existing screenshot tests with the current color, changing them again should not be a problem blocking things.

mneudert commented on plugins/Marketplace/stylesheets/plugin-details.less:
```diff
@@ -102,5 +97,249 @@
+        hr {
+            height: 1px;
+            overflow: hidden;
+            background-image: url('plugins/Marketplace/images/divider.svg');
```
Are we using an SVG to achieve consistency across all browsers? And/or to achieve the specific dash size not available with a default border?

![](https://github.com/matomo-org/matomo/assets/635801/9ef305a6-bdb5-49e8-acd6-50cb5fd964b2)
![](https://github.com/matomo-org/matomo/assets/635801/22eb2f41-2a12-435a-91de-88dd8b4815f7)

michalkleiner replied:
To achieve the wider stroke gap (as designed) that is not achievable via CSS border, though I think design team would be equally happy with the standard dashed border with a lighter colour so it had the same lightness to it. I liked the SVG approach as it helped me to learn a new thing.

mneudert commented on plugins/Marketplace/vue/src/types.ts:
```diff
@@ -0,0 +1,99 @@
+  interface Window {
+    // TODO: figure out a narrower type definition
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    iFrameResize: any;
```
There is a type package available at https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/iframe-resizer/index.d.ts.

We could copy a minimal subset from there instead of pulling in the complete definition.

michalkleiner replied:
Nice find! I've included the package in the dev dependencies. It feels like the cleanest way of keeping up to date with any changes should there be some vs keeping a custom subset.

mneudert commented on plugins/Marketplace/vue/src/types.ts:
```diff
@@ -0,0 +1,99 @@
+export type TObjectArray = TObject[] | [] | null;
+export type TNumberOrString = string | number | null;
+
+export interface IPluginShopVariation {
```
Are we planning to go with the naming convention of `interface I` and `type T`? Haven't seen that around somewhere else yet :thinking:

michalkleiner replied:
Yes, there's no precedent yet, but I think it's a good practice and it allows everyone to see at first glance what kind of definition it is compared to a name that doesn't have that indication. Happy to be overruled on this if the team feels strongly against that.

michalkleiner replied:
You can also see that I kept the interface defined for the component without the prefix for consistency with other components and only used the `I` and `T` prefixes for partials defining the main component interface.

mneudert replied:
I would only say we should align on the direction and have that decision enforced ([eslint](https://github.com/typescript-eslint/typescript-eslint/blob/v7.7.1/packages/eslint-plugin/docs/rules/naming-convention.mdx#enforce-that-type-parameters-generics-are-prefixed-with-t)) as much as possible.

Otherwise it may get a wild mix of everything and there are no real arguments to have one side chosen over the other one except personal preference.

Do @sgiehl / @caddoo have an opinion on that one?

michalkleiner replied:
If this was the only thing blocking the merge we can merge it as is and come back to it later should we chose to change the approach.

mneudert replied:
Not blocking anything. Will create an internal ticket so we can eventually align on the direction to take.

mneudert commented on plugins/Marketplace/vue/src/PluginDetailsModal/PluginDetailsModal.vue:
```diff
@@ -0,0 +1,508 @@
+            {{ translate('Marketplace_MultiServerEnvironmentWarning') }}
+          </div>
+          <div v-else-if="!isAutoUpdatePossible" class="alert alert-warning">
+            <!-- eslint-disable-next-line max-len-->
+            {{ translate('Marketplace_AutoUpdateDisabledWarning', '\'[General]enable_auto_update=1\'', '\'config/config.ini.php\'') }}
```
Don't know it that is really more readable, but it should let us get rid of the eslint comment:

```suggestion
            {{
              translate(
                'Marketplace_AutoUpdateDisabledWarning',
                '\'[General]enable_auto_update=1\'',
                '\'config/config.ini.php\''
              )
            }}
```

The other occurrences probably require a computation to not look like someone had a sticky return key :thinking:

michalkleiner replied:
Thanks, will tweak this.

I think it's ok to have the the eslint-disable comment for some exceptions (such as the long regular expression) where breaking the code onto multiple lines or using some other techniques would make it less readable or harder to understand. Happy to go with what the team thinks collectively is best.

mneudert commented on plugins/Marketplace/stylesheets/plugin-details.less:
```diff
@@ -102,5 +97,249 @@
+        justify-content: space-between;
+
+        .cta-container {
+            width: 30%;
```
Can we make this `width: auto`?

Narrowing the viewport would currently remove parts of the button text, for example displaying "Start Free". If we are only displaying the matomo logo next to it we should in most languages be able to display the complete button, perhaps with a `max-width: 50%` added so it definitely does not flow into the logo?

michalkleiner replied:
I can see if we can do minimum and maximum width here, the reason I put 30 is for consistency with the card button and not to have the button too wide.
Perhaps one or two more breakpoint related widths would help better here. I'll see what can be done here.

mneudert commented on plugins/Marketplace/stylesheets/plugin-details.less:
```diff
@@ -102,5 +97,249 @@
+
+    .modal-content__header {
+        padding: 24px 24px 0;
+        height: 160px;
```
Who would have guessed, but once a plugin title breaks into a second line ("Multi Channel Conversion Attribution" is a good candidate for that) the elements below overlap one another.

Would that be reasonably prevented just removing that setting here? In combination with not removing the bottom padding (so just `padding: 24px` above) it looked okay, haven't checked in various browsers though.

michalkleiner replied:
Kept the height and adjusted the font-size and spacing on narrower screens as the height is used to calculate the height of the content.

mneudert commented on plugins/Marketplace/vue/src/PluginList/CTAContainer.vue:
```diff
@@ -108,10 +112,12 @@
-  />
+  <template v-else>
+    <MoreDetailsButton
+      v-if="!inModal"
```
We probably should add more of those `!inModal` checks. For example the Help link is opening the old modal above the new modal when clicked :see_no_evil:

michalkleiner replied:
Good catch! With the new modal, the help link doesn't even need to exist as a separate component and can just open the details modal as the whole card does. Will adjust the behaviour.

michalkleiner replied:
I've refactored the HelpLink and MoreDetailsButton components into one as the mechanism is identical and they now do the same thing just looking differently. I've added the `inModal` condition to multiple more places and now it doesn't show inside the modal any links or buttons that would open the (old) modal again.

mneudert reviewed (commented):
I like how fluffy and responsive this feels, compared to the old modals :hugs:

michalkleiner committed:
Remove unused computed property

michalkleiner committed:
Fix license spelling in variable and method names

michalkleiner committed:
Remove number of users from Marketplace as product variant selection for free trials is automatic in the backend

michalkleiner committed:
Add email validity check to author information section

michalkleiner committed:
Rearrange code to remove some of the max line length lint exceptions

michalkleiner committed:
Move component state definition to the component file

michalkleiner committed:
Unify email sanitisation for author and supporting metadata

michalkleiner committed:
Add types for iframeResizer

michalkleiner committed:
Merge HelpLink and MoreDetailsButton components into MoreDetailsAction

Both components were doing the same thing in a different form of a link (text vs button),
so these got merged with props to control the button block look and the label.

michalkleiner committed:
Adjust action links within CTA container to only appear when not presented in a modal

michalkleiner committed:
Update dist files

michalkleiner committed:
Improve responsive behaviour on smaller screens

mneudert commented on plugins/Marketplace/stylesheets/plugin-details.less:
```diff
@@ -3,11 +3,6 @@
     text-align: left;
     line-height: 20px;
 
-    > .row > .col {
```
This change should not be needed here anymore?

Removing the change should remove the visual changes of the old modals, and we can keep your other suggested change for DEV-18004.

michalkleiner replied:
Yep, I will revert this, it was a leftover from the first iteration.

michalkleiner replied:
Reverted.

mneudert commented on plugins/Marketplace/vue/src/PluginList/CTAContainer.vue:
```diff
@@ -34,7 +36,7 @@
         <span
           style="white-space:nowrap"
-        >(<HelpLink :plugin-name="plugin.name" />
+        >(<MoreDetailsAction @action="$emit('openDetailsModal')" v-if="!inModal" />
```
Here, and behind the `Marketplace_CannotInstall` text, the `<span>` needs to be have the `!inModal` check. Otherwise the braces are displayed without contents.

michalkleiner replied:
The `!inModal` check can't simply move to the span as there are potentially two links inside the brackets in those locations (help and download), so I've added an or (`||`) check for the logic for both links to the `span` so it should not display empty brackets anymore.

michalkleiner committed:
Ensure there's no empty text in CTA link actions

mneudert committed:
Update selectors for marketplace UI modal tests

mneudert committed:
Remove "min-height" to prevent huge modals on huge screens

mneudert committed:
Keep minimal UI tests for old modal

mneudert committed:
Update expected screenshots
